polaris-core/src/test/java/org/apache/polaris/service/storage/StorageLocationTest.java

```diff
@@ -19,17 +19,41 @@
 package org.apache.polaris.service.storage;
 
 import org.apache.polaris.core.storage.StorageLocation;
-import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
 public class StorageLocationTest {
 
   @Test
-  public void testOfDifferentPrefixes() {
-    StorageLocation StandardLocation = StorageLocation.of("file:///path/to/file");
+  public void testAwsLocations() {
+    StorageLocation baseStorageLocation = StorageLocation.of("s3://path/to/file");
+    StorageLocation testStorageLocation = StorageLocation.of("s3://path/to/////file");
+    Assertions.assertEquals(testStorageLocation, baseStorageLocation);
+    testStorageLocation = StorageLocation.of("s3://////path/////to///file");
+    Assertions.assertEquals(testStorageLocation, baseStorageLocation);
+
+    testStorageLocation = StorageLocation.of("s3://////path'/////*to===///\"file");
+    Assertions.assertEquals("s3://path'/*to===/\"file", testStorageLocation.toString());
```

Comment:
nit: in this URI `path'` is not a "path" by "authority" per [RFC 3986](https://www.rfc-editor.org/rfc/rfc3986)